<?xml version="1.0"?>
<launch>
  <!-- By default we do not overwrite the URDF. Change the following to true to change the default behavior -->
  <!-- The name of the parameter under which the URDF is loaded -->
  <arg name="robot_description" default="robot_description"/>
  <arg name="plugin" default="tesseract_collision/BulletDiscreteBVHManager"/>
  <arg name="monitor_links" default=""/>

  <!--
  FIRST = 0,   /**< Return at first contact for any pair of objects */
  CLOSEST = 1, /**< Return the global minimum for a pair of objects */
  ALL = 2,     /**< Return all contacts for a pair of objects */
  LIMITED = 3  /**< Return limited set of contacts for a pair of objects */
  -->
  <arg name="contact_test_type" default="2"/>
  <arg name="contact_distance"/>
  <arg name="publish_environment" default="false"/>
  <arg name="publish_markers" default="false"/>


  <node pkg="tesseract_monitoring" type="tesseract_monitoring_contacts_node" name="tesseract_monitoring_contacts">
    <param name="robot_description" type="string" value="$(arg robot_description)"/>
    <param name="plugin" type="string" value="$(arg plugin)"/>
    <param name="contact_distance" value="$(arg contact_distance)"/>
    <param name="monitor_links" value="$(arg monitor_links)"/>
    <param name="contact_test_type" value="$(arg contact_test_type)"/>
    <param name="publish_environment" value="$(arg publish_environment)"/>
    <param name="publish_markers" value="$(arg publish_markers)"/>
  </node>

</launch>
